revolv/payments/models.py

```diff
@@ -2,63 +2,231 @@
-           revolv the money that is represented in the Payment that has this PaymentInstrumentType.
+
+class PaymentType(models.Model):
+    """
+    Abstraction for a payment. A payment type is defined as anything that can
+    be used to create a payment (for example, Paypal, check, or any other
+    payment service). Types of payment that the RE-volv app currently supports
+    are as follows:
+
+    1. "paypal": a payment that was made via a credit card processed through
+        the paypal REST sdk. TODO: the distinction between an actual payment
+        through a paypal account and a payment made with credit card but
+        processed by paypal.
+    2. "check": a record or a payment that was made directly to a RE-volv
+        administrator via paper check. These must be entered into the app via
+        the UI in the administrator dashboard page in order for our database to
+        know about them.
+    3. "reinvestment": this is a payment that was allocated to a new project as
+        part the revolving funding model - the money originally came from an
+        organic donation to an already completed project, and then the
+        community repayed RE-volv the money that is represented in the Payment
+        that has this PaymentType.
     """
     name = models.CharField(max_length=80, unique=True)
 
-    objects = PaymentInstrumentTypeManager()
+    objects = PaymentTypeManager()
+
+
+class RepaymentManager(models.Manager):
+    """
+    Manager for repayments.
+    """
+
+    def repayments(self, **kwargs):
+        """
+        :return:
+            Repayments associated with this user, project, and/or
+            AdminRepayment.
+        """
+        if kwargs.get('queryset') is None:
+            queryset = super(RepaymentManager, self).get_queryset()
+        if kwargs.get('user'):
+            queryset = queryset.filter(user=kwargs['user']).order_by('created_at')
+        if kwargs.get('project'):
+            queryset = queryset.filter(project=kwargs['project']).order_by('created_at')
+        if kwargs.get('admin_repayment'):
+            queryset = queryset.filter(admin_repayment=kwargs['admin_repayment']).order_by('created_at')
+        return queryset
+
+
+class Repayment(models.Model):
+    """
+    Abstraction for a repayment to a particular RevolvUserProfile.
+
+    A Repayment represents the proportional amount of money that the associated
+    user is repayed for an AdminRepayment to a project. In other words, for user
+    U, project P, and AdminRepayment R, the amount of the Repayment will be:
+
+        ((U's donation to P) / (Total organic donations to P)) * R.amount
+
+    When a Repayment is generated/deleted, we automatically increment/decrement
+    the pool of reinvestable money in the related user.
+
+    Repayments are generated automatically when an AdminRepayment is created.
+    Repayments should never be created manually!
+
+    ::Signals::
+    post_save
```

Comment:
Same thing here as with AdminReinvestment, a Repayment should always only be saved once, right? Because if it's saved multiple times, it will increment the reinvest_pool multiple times. If you decide to just use the created kwarg like I suggested earlier, you could have a test to make sure the reinvest_pool is only incremented once when you have multiple times :)